reports/platopsapps/version_utility.py

```diff
@@ -0,0 +1,147 @@
+import requests
+import logging
+import re
+from bs4 import BeautifulSoup
+
+logging.basicConfig(
+    level=logging.INFO,
+    format='%(asctime)s [%(levelname)s] %(message)s',
+    datefmt='%Y-%m-%d %H:%M:%S'
+)
+
+def get_semvar(version_number_str):
+    version = re.search(r'(\d+\.\d+\.\d+)', version_number_str)
+    return version.group(1)
+
+def get_patch_version(version_number_str):
+    return int(version_number_str[version_number_str.rfind('.') + 1:])
+
+def get_minor_version(version_number_str):
+    return int(version_number_str.split('.')[1])
+
+def get_major_version(version_number_str):
+    return int(version_number_str[0: version_number_str.find('.')])
+
+def compare_versions(current_version, latest_version, service_name):
+    """
+    This function will compare to semantic versions to check if the current version is older than the latest and return a status based on the outcome.
+    The function expects 3 values to be supplied in string format:
+        - current version -  a semvar formatted version string (e.g. 1.2.3).
+        - latest version -  a semvar formatted version string (e.g. 1.2.3).
+        - service_name - a string containing the name of the service for which the versions are being checked, used in log output only.
+    """
+
+    try:
+        logging.info(f"{service_name} current version: {current_version}")
+        logging.info(f"{service_name} latest version: {latest_version}\n")
+
+        latest_major = get_major_version(latest_version)
+        current_major = get_major_version(current_version)
+        latest_minor = get_minor_version(latest_version)
+        current_minor = get_minor_version(current_version)
+        latest_patch = get_patch_version(latest_version)
+        current_patch = get_patch_version(current_version)
```

Comment:
Think it's useful to check these are all set before  the below if statement? In case external URL is flaky etc - maybe causes issues in DB with empty data / an exception to be thrown comparing things that aren't set

Reply:
I see you have https://github.com/hmcts/version-reporter-services/pull/170/files#diff-e47cca963eb4b74fd032b5a83823b587fe8fe22f7497c3fd9c966260d5198798R65-R71 but not sure how this works with data into db